ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/multisplitpanel/menu/MenuWidget.java

```diff
@@ -106,24 +108,41 @@ public void onClose(CloseEvent<PopupPanel> event) {
 
   public void showList() {
     int x = getAbsoluteLeft() + getOffsetWidth() - 6;
-    int y = getAbsoluteTop() + 19;
-
     popupPanel.show();
     popupPanel.getElement().getStyle().setProperty("position", "absolute");
     popupPanel.getElement().getStyle().clearProperty("left");
     popupPanel.getElement().getStyle().setProperty("right", "calc(100% - " + x + "px");
-    popupPanel.getElement().getStyle().setProperty("top", "" + y + "px");
+    fixPositionIfNeed();
   }
 
   @Override
   public void addListItem(MenuItem menuItem) {
+    itemsInMenu++;
     menuItem.setDelegate(itemDelegate);
     listPanel.add(menuItem);
+    fixPositionIfNeed();
   }
 
   @Override
   public void removeListItem(MenuItem menuItem) {
+    itemsInMenu--;
     listPanel.remove(menuItem);
+    fixPositionIfNeed();
+  }
+
+  /**
+   * Fix top value if need. Need in case if bottom part of menu not display
+   */
+  private void fixPositionIfNeed() {
```

Comment:
Let's name it as `adjustTopPosition`? I think this name is clearer and better reflects what the method does.

Reply:
+1 thx, I tried to found this name